utils/s2n_mem.c

```diff
@@ -53,16 +53,26 @@ int s2n_realloc(struct s2n_blob *b, uint32_t size)
         return s2n_free(b);
     }
 
+    /* blob already has space for the request */
     if (size < b->allocated) {
         b->size = size;
         return 0;
     }
 
```

Comment:
Would we better doing something like:

    if (!use_mlock) {
        b->data = realloc(b->data ...)
        madvice(DONT_DUMP) ...
        return 0;

here? There's a lot of logic in this function to duplicate what realloc may do more efficiently.